packages/editor-ui/src/composables/useNodeHelpers.ts

```diff
@@ -373,8 +373,15 @@ export function useNodeHelpers() {
 			node.credentials !== undefined
 		) {
 			const stored = credentialsStore.getCredentialsByType(nodeCredentialType);
-
-			if (selectedCredsDoNotExist(node, nodeCredentialType, stored)) {
+			// Prevents HTTP Request node from being unusable if a sharee does not have direct
```

Comment:
@krynble   I'm not sure but it feels like this case is handled a couple of lines down and the problem is rather with the condition of this block.

Doesn't the `selectedCredsAreUnusable` function do something similar?

Reply:
Actually the block of the next condition is doing the same
Couldn't they be merged?

Reply:
From my understanding, the current block is checking whether the selected credential does not exist.

But the "exist" part is wrong from what I can tell, because it's ignoring the `usedCredentials` attribute that highlights credentials owned by other sharees of the workflow.

The next block seems to be looking at credentials that are set, but are of invalid type (say for whatever reason a credential to Hubspot was set to a Pipedrive node).

Then below there's a `for` loop that I don't quite understand what it does for all cases but after debugging I was able to spot a difference in behavior for regular nodes (dedicated to a specific service) vs http request node, hence the change I've made.

Reply:
I just saw these identical blocks and thought maybe there is a chance to simplify
![CleanShot 2024-03-08 at 12 50 00](https://github.com/n8n-io/n8n/assets/5410822/234b1440-0a68-4f57-a49b-ca38ceeb5b42)

Reply:
Ah yes, there are a few similarities between those blocks, but since they have slightly different checks, for the sake of clarity (which is not great I know, but I would like to avoid making it worse) I think they can remain separate.